### ISSUE TITLE
🌸 Cherry pick request for 16383 into 1529691528033 (Pending)

### ISSUE THREAD

**Issue by bradfrizzell** [ampproject/amphtml#16385]
**GitHub issue your cherry pick is fixing:**

Issue #16384

**PR that you are requesting a cherry pick for:**

*(Put N/A if you do not yet have a PR with a fix; edit this issue to add it when the PR is ready.)*

PR #16383

**Release(s) you requesting this cherry pick into:**

*Release issues can be found at https://github.com/ampproject/amphtml/labels/Type%3A%20Release*

Production Release? YES  If yes, Type: Release Issue #16025

Canary release? YES  If yes, Type: Release Issue #16178,

*Why does this issue meet the [cherry pick criteria](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md#cherry-pick-criteria)?  Be specific.*
This will break RTC and throw errors on any page that is using RTC, but has an empty URLs config, which is likely a vast amount of RTC. 

*Assign this issue to the current TL (cramforce) if you have permission to, otherwise leave this cc line in.*

/cc @cramforce

**Comment by cramforce** [comment]
This is a cherry-pick into the current canary correct?

**Comment by jridgewell** [comment]
What's the impact here? I don't see any error reports for this yet.

**Comment by bradfrizzell** [comment]
This needs to be merged into each release that has https://github.com/ampproject/amphtml/pull/15951

I'm surprised that we aren't seeing error reports for this yet, as I believe it should affect any page that doesn't have a setting for "urls" in their RTC config. Keith found this bug by seeing it happen randomly in the wild

**Comment by jridgewell** [comment]
I see, you've surrounded it in a [try-catch](https://github.com/bradfrizzell/amphtml/blob/e3abeff1b4414b698db1da805708afef9eb00417/extensions/amp-a4a/0.1/real-time-config-manager.js#L518-L521). We won't get error reports for this.

**Comment by cramforce** [comment]
Depending on impact it might make sense to only merge into new release.

**Comment by bradfrizzell** [comment]
the new prod or new canary?

**Comment by jridgewell** [comment]
I haven't cut new canary, so it'll be in there no matter what.

The question is do we add this to the already in-progress 1% patch.

**Comment by bradfrizzell** [comment]
We are seeing a substantial shift in our metrics that we believe this issue accounts for

**Comment by cramforce** [comment]
So, this is approved for both stable and canary, but I'm a bit puzzled how we keep NOT catching these in our canary process?
CC @keithwrightbos

**Comment by bradfrizzell** [comment]
the last issue was not caught specifically because it only affected canonical traffic, so the lack of an unversioned canary meant we couldn't have found it this way.

**Comment by cramforce** [comment]
CC @rsimha
I think plans for ^^^^ have stalled, but need to be picked up eventually.

**Comment by jridgewell** [comment]
PR merged for canary release. Waiting on @cramforce for cherry-pick approval.

**Comment by cramforce** [comment]
Approval is above, hidden in my short rant :)

**Comment by jridgewell** [comment]
👍. Moving to https://github.com/ampproject/amphtml/issues/16388.

**Comment by rsimha** [comment]
Just chatted with @erwinmombay, and neither of us fully understand what "unversioned canary" means.

@bradfrizzell @cramforce Could you elaborate?

**Comment by jridgewell** [comment]
He means `cdn.ampproject.org/v0.js` should serve the canary version in 1% of cases. Right now, `/v0.js` only serves the canary version if you've opted-in.

**Comment by bradfrizzell** [comment]
right, the issue is that right now the only canary we serve is for pages served via the cdn. there is no canarying of cdn.ampproject.org/v0.js for canonical pages
